src/twitter.d.ts

```diff
@@ -0,0 +1,64 @@
+declare module 'twitter-v2' {
+  export interface ApplicationCredentials {
+    consumer_key: string;
+    consumer_secret: string;
+    bearer_token?: string;
+  }
+
+  export interface UserCredentials {
+    consumer_key: string;
+    consumer_secret: string;
+    access_token_key: string;
+    access_token_secret: string;
+  }
+
+  export type CredentialsArgs = ApplicationCredentials | UserCredentials;
+
+  export interface RequestParameters {
+    [key: string]: any;
+  }
+
+  export interface TwitterResponse<T extends any> {
+    data: T;
+  }
```

Comment:
The response varies by endpoint and data returned. Other fields that show up regularly are `meta` and `errors`. In my mind there are two ways to deal with this:

1. Create a super-interface that contains every field twitter can return
2. Return any

Personally I don't feel strongly and am happy to lean on Typescript convention here. Up to you. It's worth noting that option 1 will be difficult with generics.

Reply:
Gotcha. Every endpoint I was seeing returned objects under the `data` field, but I guess I didn't look far enough (https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/api-reference/post-tweets-search-stream-rules#success). I will allow the endpoints to be generically typed (but remove this `TwitterResponse` interface) so that consumers can easily setup their interfaces/types as necessary. Leaving it as `any` will require them to do an additional cast after the response comes back.